infer/src/base/Config.ml

```diff
@@ -497,6 +498,14 @@ and bootclasspath =
     ~exes:CLOpt.[Toplevel; Java]
     "Specify the Java bootclasspath"
 
+and bo_debug =
+  CLOpt.mk_int ~default:0 ~long:"bo-debug"
+    ~exes:CLOpt.[Toplevel] "Debug mode for buffer-overrun checker"
+
+and bo_filtering =
```

Comment:
The two best files to look at are localise.ml (see the `Buckets` module) and errdesc.ml (see the `explain` functions and their usages from other modules. 

Although things are a bit disorganized, the typical pattern is to assign a warning to a bucket from B1 (highest confidence) to B5 (highest confidence) while creating a `Localise.error_desc`. The actual filtering (which shouldn't need to be changed) happens in `Exceptions.recognize_exception`. By default, buckets B1 and B2 are reported, and higher buckets are filtering unless the `--no-filtering` flag is passed.